### PR TITLE
Update rule S4275 (no-accessor-field-mismatch): cover cases from ESLint's getter-return

### PR DESCRIPTION
**Pull request by francoismora**
Fixes #3702 
RSPEC: SonarSource/RSPEC#1643

### CODE REVIEW

File: tests/linting/eslint/rules/no-accessor-field-mismatch.test.ts [hunk 1]
```diff
@@ -44,38 +44,54 @@ function invalid(code: string) {
...
+
 ruleTester.run('Getters and setters should access the expected fields', rule, {
   valid: [
     {
       code: `
         class OK {
           private x: string;
           private _y = "hello";
-        
+
```
Comment (yassin-kammoun-sonarsource) [inline tests/linting/eslint/rules/no-accessor-field-mismatch.test.ts:70]: I am surprised to see these differences. Was it the pre-commit hook that did that?

**Review by yassin-kammoun-sonarsource — APPROVED** [review]
LGTM! Nice work! We might need to update the RSPEC description in case you make some changes in the corresponding RPSEC PR.

File: tests/linting/eslint/rules/no-accessor-field-mismatch.test.ts (hunk 1 above)
Comment (francoismora) [reply tests/linting/eslint/rules/no-accessor-field-mismatch.test.ts:70]: I think it comes from the IDE comment/uncomment operations I used when debugging unit tests.

Comment (yassin-kammoun-sonarsource) [reply tests/linting/eslint/rules/no-accessor-field-mismatch.test.ts:70]: 👍

**Pull request merged**